src/test/checkqueue_tests.cpp

```diff
@@ -372,29 +381,40 @@ BOOST_AUTO_TEST_CASE(test_CheckQueueControl_Locks)
     }
     {
         boost::thread_group tg;
-        std::atomic<bool> has_lock {false};
-        std::atomic<bool> done {false};
-        std::atomic<bool> done_ack {false};
-        tg.create_thread([&, queue]{
-                CCheckQueueControl<FakeCheck> control(queue.get());
-                has_lock = true;
-                MilliSleep(1000);
-                done = true;
-                // Wait until the done is acknowledged
-                while (!done_ack) {
-                }
-                });
-        // Wait for thread to get the lock
-        while (!has_lock) {}
-        bool fails2 = false;
-        // Agressively check that we can never aquire the lock, once
-        // the thread has stopped check that we never got it.
-        while (!done && !fails2) {
-            fails2 = queue->ControlMutex.try_lock();
+        std::mutex m;
+        bool has_lock {false};
```

Comment:
Going to ignore this, I don't think it makes it more clear (to me, it's easier to debug several variables). If someone disagrees strongly, will change.